ingestion/src/metadata/ingestion/source/pipeline/gluepipeline/metadata.py

```diff
@@ -210,3 +325,49 @@ def yield_pipeline_lineage_details(
         """
         Get lineage between pipeline and data sources
         """
+        try:
+            for job in self.job_name_list:
+                lineage_enities = self.get_lineage_details(job)
+                pipeline_fqn = fqn.build(
+                    metadata=self.metadata,
+                    entity_type=Pipeline,
+                    service_name=self.context.get().pipeline_service,
+                    pipeline_name=self.context.get().pipeline,
+                )
+
+                pipeline_entity = self.metadata.get_by_name(
+                    entity=Pipeline, fqn=pipeline_fqn
+                )
+
+                lineage_details = LineageDetails(
+                    pipeline=EntityReference(
+                        id=pipeline_entity.id.root, type="pipeline"
+                    ),
+                    source=LineageSource.PipelineLineage,
+                )
+                for source in lineage_enities.get("sources"):
+                    for target in lineage_enities.get("targets"):
+                        yield Either(
+                            right=AddLineageRequest(
+                                edge=EntitiesEdge(
+                                    fromEntity=EntityReference(
+                                        id=source[0].id,
+                                        type=source[1],
```

Comment:
not sure if I like this approach just to find the, instead of storing the entire entity and type in this dict, instead let's just store the entity reference instead of get_by_name we can use [get_entity_reference](https://github.com/open-metadata/OpenMetadata/blob/53cdd34391e5a541d153d89786dafcb65365e900/ingestion/src/metadata/ingestion/ometa/ometa_api.py#L359) and use it directly here